api/src/transactions.rs

```diff
@@ -1182,14 +1182,19 @@ impl TransactionsApi {
         // Simulate transaction
         let state_view = self.context.latest_state_view_poem(&ledger_info)?;
         let move_resolver = state_view.as_move_resolver();
-        let (_, output_ext) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
+        let (_, vm_output) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
         let version = ledger_info.version();
 
-        // Apply transaction outputs to build up a transaction
-        // TODO: while `into_transaction_output_with_status()` should never fail
-        // to apply deltas, we should propagate errors properly. Fix this when
-        // VM error handling is fixed.
-        let output = output_ext.into_transaction_output(&move_resolver);
+        let output = match vm_output.try_materialize(&move_resolver) {
+            Ok(vm_output) => vm_output.output_with_materialized_deltas(vec![]),
```

Comment:
doesn't try_materialize already merge deltas converted to write-ops? why do we have to call output_with_materialized deltas with an empty vector right after?

Reply:
This is because of parallel execution actually: we create delta writes and we apply them to output, right? But the output already has deltas. This function used to be this one:

```rust
pub fn output_with_delta_writes(
        self,
        delta_writes: Vec<(StateKey, WriteOp)>,
) -> TransactionOutput {
  ...
}
```